app/helpers/pagelets_helper.rb

```diff
@@ -3,33 +3,39 @@ def pagelets_for(mountpoint)
     Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{action_name}", mountpoint)
   end
 
-  def render_pagelets_for(mountpoint, opts = {})
+  def render_pagelets_for(mountpoint, subject, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
-      result += render_pagelet(pagelet, opts)
+      result += render_pagelet(pagelet, subject, opts)
     end
     result.html_safe
   end
 
-  def render_tab_content_for(mountpoint, opts = {})
+  def render_tab_content_for(mountpoint, subject, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
+      next unless pagelet.onlyif.call subject
       result += "<div id='#{pagelet.id}' class='tab-pane'>"
-      result += render_pagelet(pagelet, opts)
+      result += render_pagelet(pagelet, subject, opts)
       result +=  "</div>"
     end
     result.html_safe
   end
 
-  def render_tab_header_for(mountpoint, opts = {})
+  def render_tab_header_for(mountpoint, subject, opts = {})
     result = ""
     pagelets_for(mountpoint).each do |pagelet|
+      next unless pagelet.onlyif.call subject
       result += "<li><a href='##{pagelet.id}' data-toggle='tab'>#{_(pagelet.name)}</a></li>"
     end
     result.html_safe
   end
 
-  def render_pagelet(pagelet, opts)
-    render(pagelet.partial, opts.merge!({ :pagelet => pagelet })).html_safe
+  def render_pagelet(pagelet, subject, opts = {})
```

Comment:
This is listed as a plugin API on the wiki, it's not a good idea to change how it works, is there a better way to keep this working, e.g. using a consistent key in `opts`?